src/vs/workbench/api/common/extHostQuickDiff.ts

```diff
@@ -33,10 +34,10 @@ export class ExtHostQuickDiff implements ExtHostQuickDiffShape {
 			.then<UriComponents | null>(r => r || null);
 	}
 
-	async registerQuickDiffProvider(quickDiffProvider: vscode.QuickDiffProvider, label: string, rootUri?: vscode.Uri): Promise<vscode.Disposable> {
+	registerQuickDiffProvider(selector: vscode.DocumentSelector, quickDiffProvider: vscode.QuickDiffProvider, label: string, rootUri?: vscode.Uri): vscode.Disposable {
 		const handle = ExtHostQuickDiff.handlePool++;
 		this.providers.set(handle, quickDiffProvider);
-		await this.proxy.$registerQuickDiffProvider(handle, label, rootUri);
+		this.proxy.$registerQuickDiffProvider(handle, DocumentSelector.from(selector), label, rootUri);
```

Comment:
you might wanna pass a uriTransformer to the `from` call. That ensures that selectors using certain schemes are correct

Reply:
Thanks for the heads up! https://github.com/microsoft/vscode/pull/171614